omnigibson/systems/micro_particle_system.py

```diff
@@ -1540,44 +1542,58 @@ def clothify_mesh_prim(cls, mesh_prim, remesh=True, particle_distance=None):
-            tmp_name = f"{mesh_prim.GetName()}_{datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S')}"
+            tmp_name = str(uuid.uuid4())
             tmp_dir = os.path.join(tempfile.gettempdir(), tmp_name)
             tmp_fpath = os.path.join(tmp_dir, f"{tmp_name}.obj")
             Path(tmp_dir).mkdir(parents=True, exist_ok=True)
             tm.export(tmp_fpath)
-            ms = pymeshlab.MeshSet()
-            ms.load_new_mesh(tmp_fpath)
 
-            # Re-mesh based on @particle_distance - distance chosen such that at rest particles should be just touching
-            # each other. The 1.5 magic number comes from the particle cloth demo from omni
-            # Note that this means that the particles will overlap with each other, since at dist = 2 * contact_offset
-            # the particles are just touching each other at rest
+            # Start with the default particle distance
             particle_distance = cls.particle_contact_offset * 2 / (1.5 * np.mean(mesh_prim.GetAttribute("xformOp:scale").Get())) \
                 if particle_distance is None else particle_distance
-            avg_edge_percentage_mismatch = 1.0
-            iters = 0
-            # Loop re-meshing until average edge percentage is within error threshold or we reach the max number of tries
-            while avg_edge_percentage_mismatch > m.CLOTH_REMESHING_ERROR_THRESHOLD:
-                ms.meshing_isotropic_explicit_remeshing(iterations=5, targetlen=pymeshlab.AbsoluteValue(particle_distance))
-                avg_edge_percentage_mismatch = abs(1.0 - particle_distance / ms.get_geometric_measures()["avg_edge_length"])
-                iters += 1
-                if iters > 5:
+
+            # Repetitively re-mesh at lower resolution until we have a mesh that has less than MAX_CLOTH_PARTICLES vertices
+            for _ in range(3):
+                ms = pymeshlab.MeshSet()
+                ms.load_new_mesh(tmp_fpath)
+
+                # Re-mesh based on @particle_distance - distance chosen such that at rest particles should be just touching
+                # each other. The 1.5 magic number comes from the particle cloth demo from omni
+                # Note that this means that the particles will overlap with each other, since at dist = 2 * contact_offset
+                # the particles are just touching each other at rest
+
+                avg_edge_percentage_mismatch = 1.0
+                # Loop re-meshing until average edge percentage is within error threshold or we reach the max number of tries
+                for _ in range(5):
+                    if avg_edge_percentage_mismatch <= m.CLOTH_REMESHING_ERROR_THRESHOLD:
+                        break
+
+                    ms.meshing_isotropic_explicit_remeshing(iterations=5, targetlen=pymeshlab.AbsoluteValue(particle_distance))
+                    avg_edge_percentage_mismatch = abs(1.0 - particle_distance / ms.get_geometric_measures()["avg_edge_length"])
+                else:
```

Comment:
all the else statements in this function seem weird, can you double check? the logic is a bit hard to follow.

Reply:
just checked - the else statements run when the loop isnt broken out of via a `break` statement, which we do in the success cases. So in the failure cases we go into the elses.